### PR TITLE
refactor: consolidate modal requests

### PR DESCRIPTION
**Pull request by gillycheesesteak**

### CODE REVIEW

File: src/modal/parts/Calculator/DE/Calculator.jsx [hunk 1]
```diff
@@ -13,14 +14,20 @@ const Calculator = () => {
             <div className="calculator">
                 <h3 className="calculator__title">Monatliche Raten berechnen</h3>
                 <form className="calculator__form" onSubmit={submit}>
-                    <input className="calculator__input" value={value} onInput={changeInput} />
+                    <input
+                        className="calculator__input"
+                        value={value}
+                        onInput={changeInput}
+                        disabled={isLoading}
```
Comment (Seavenly) [inline src/modal/parts/Calculator/DE/Calculator.jsx]: Can this be removed for accessibility?

File: src/modal/parts/Calculator/DE/Calculator.jsx [hunk 2]
```diff
@@ -13,14 +14,20 @@ const Calculator = () => {
...
+                        style={{ opacity: isLoading ? '0.5' : '1' }}
+                    />
                     <p className="calculator__instructions">
                         {!terms.error &&
                             terms.formattedMinAmount &&
                             terms.formattedMaxAmount &&
                             `Geben Sie einen Betrag zwischen ${terms.formattedMinAmount}€ und ${terms.formattedMaxAmount}€ ein.`}
                     </p>
-                    <Button size="md" type="submit">
+                    <Button size="md" type="submit" disabled={isLoading}>
```
Comment (Seavenly) [inline src/modal/parts/Calculator/DE/Calculator.jsx]: Same as above.

File: src/modal/parts/Calculator/US/Calculator.jsx [hunk 3]
```diff
@@ -13,8 +13,15 @@ const Calculator = () => {
...
                 <h2 className="calculator__title">Enter a purchase amount to calculate your monthly Easy Payments.</h2>
                 <form className="calculator__form" onSubmit={submit}>
-                    <input className="calculator__input" type="text" value={value} onInput={changeInput} />
-                    <Button className="calculator__button" type="submit" secondary>
+                    <input
+                        className="calculator__input"
+                        type="text"
+                        value={value}
+                        onInput={changeInput}
+                        disabled={isLoading}
```
Comment (Seavenly) [inline src/modal/parts/Calculator/US/Calculator.jsx]: Same as above.

File: src/modal/lib/hooks/calculator.js [hunk 4]
```diff
@@ -1,10 +1,35 @@
...
 import { useXProps } from './helpers';
 import { ServerContext } from '../context';
 import { request } from '../../../utils';
 
+const termsCache = new Map();
+
+const getCacheKey = params =>
+    ['amount', 'country', 'client_id', 'payer_id', 'merchant_id'].map(key => params[key]).join('_');
+
+const termsFetcher = async (params, csrf) => {
```
Comment (Seavenly) [inline src/modal/lib/hooks/calculator.js]: Could we use the existing `memoize` helper function to achieve the same thing as creating another type of cache here? Or we could possibly refactor a bit and use `useMemo` to store the request promise and read it when needed.

Comment (gillycheesesteak) [reply src/modal/lib/hooks/calculator.js]: I couldn't think of a way to use the existing memoize while also using the terms that are server-rendered and having them be cached as well. Thoughts?

**Pull request merged**